src/collection-view.js

```diff
@@ -28,6 +28,7 @@ Marionette.CollectionView = Marionette.View.extend({
 
     Marionette.View.apply(this, arguments);
 
+    this.on('before:show', this._onBeforeShowCalled);
```

Comment:
going to go on record to say this is a little weird in the `constructor`, but it's consistent with what we're doing with `show`. Maybe we re-evaluate or at a minimum add a comment describing the rationale.